src/main/java/com/tterrag/registrate/builders/ItemBuilder.java

```diff
@@ -108,12 +109,29 @@ public static <T extends Item, P> ItemBuilder<T, P> create(AbstractRegistrate<?>
 
     @Nullable
     private NonNullSupplier<Supplier<ItemColor>> colorHandler;
-    @Nullable private NonNullSupplier<? extends CreativeModeTab> tab = null;
-    private boolean registeredTabEvent = false;
+    private Map<NonNullSupplier<? extends CreativeModeTab>, Consumer<AbstractRegistrate.CreativeModeTabModifier>> creativeModeTabs = Maps.newHashMap();
 
     protected ItemBuilder(AbstractRegistrate<?> owner, P parent, String name, BuilderCallback callback, NonNullFunction<Item.Properties, T> factory) {
         super(owner, parent, name, callback, ForgeRegistries.Keys.ITEMS);
         this.factory = factory;
+
+        // delegate creative mode tab modification callback registration
```

Comment:
This is a nice explanation, but these important notes about how the tab methods work should be in javadoc on those methods.